### PR TITLE
fix(ClientRequest): destroy socket when destroying IncomingMessage

### LINKED ISSUE #449: Propagate socket error event on res.destroy
[According to docs](https://nodejs.org/api/http.html#messagedestroyerror), `response.destroy(err)` emits `error` event.:

> If error is provided, an 'error' event is emitted on the socket and error is passed as an argument to any listeners on the event.

https://github.com/nodejs/node/blob/main/lib/internal/streams/destroy.js#L136

--- 
Nock [handles ](https://github.com/nock/nock/blob/feaa66fa64d24f95937ef759cdd5a7ca07646f1c/lib/intercepted_request_router.js#L125) this by overriding the Socket class and propagate the event to the request.

@kettanaito 
I tried to open a PR for this but couldn't understand how to do this, but I believe I'll be able to with your direction.

### PR DESCRIPTION
**Pull request by kettanaito**
- Fixes #449

### CODE REVIEW

File: src/interceptors/ClientRequest/MockHttpSocket.ts [hunk 1]
```diff
@@ -153,6 +158,12 @@ export class MockHttpSocket extends MockSocket {
     }
 
     const socket = this.createConnection()
+
+    // If the developer destroys the socket, destroy the original connection.
+    this.once('error', (error) => {
```
Comment (kettanaito) [inline src/interceptors/ClientRequest/MockHttpSocket.ts:163]: Added this: if the socket gets destroyed by the developer, abort the pending actual socket connection.

File: src/interceptors/ClientRequest/MockHttpSocket.ts [hunk 2]
```diff
@@ -308,6 +319,11 @@ export class MockHttpSocket extends MockSocket {
     serverResponse.removeHeader('connection')
     serverResponse.removeHeader('date')
 
+    // If the developer destroy the socket, gracefully destroy the response.
+    this.once('error', () => {
```
Comment (kettanaito) [inline src/interceptors/ClientRequest/MockHttpSocket.ts:323]: The same for the mocked responses: if the socket gets destroyed, abort the pending server response.

**Review by mikicho — APPROVED** [review]
Great!

File: src/interceptors/ClientRequest/index.test.ts [hunk 3]
```diff
@@ -60,25 +64,23 @@ it('forbids calling "respondWith" multiple times for the same request', async ()
...
-    requestEmitted.resolve()
-    await sleep(10_000)
+    await sleep(1_000)
     request.respondWith(new Response())
   })
 
   const abortController = new AbortController()
   const request = http.get(requestUrl, { signal: abortController.signal })
 
-  await requestEmitted
```
Comment (kettanaito) [inline src/interceptors/ClientRequest/index.test.ts:73]: Awaiting the request listener, for some reason, breaks this test. It's not crucial for the thing we are trying to check so I'm omitting it.

**Comment by kettanaito** [comment]
## Released: v0.32.1 🎉

This has been released in v0.32.1!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.32.1)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.32.1)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**